pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java

```diff
@@ -160,6 +171,52 @@ private static void generateBrokerBasicMetrics(PulsarService pulsar, SimpleTextO
                 clusterName, Collector.Type.GAUGE, stream);
     }
 
+
+    private static void generateLedgerOffloaderMetrics(PulsarService pulsar, SimpleTextOutputStream stream,
+                                                       boolean includeTopicMetrics) {
+        List<Metrics> metricList = new LinkedList<>();
+        if (includeTopicMetrics) {
+            pulsar.getBrokerService().getMultiLayerTopicMap().forEach((namespace, bundlesMap) -> {
+                bundlesMap.forEach((bundle, topicsMap) -> {
+                    topicsMap.forEach((topicName, topic) -> {
+                        if (topic instanceof PersistentTopic) {
+                            //check offlaoder
+                            PersistentTopic persistentTopic = (PersistentTopic) topic;
+                            LedgerOffloader offloader =
+                                    persistentTopic.getManagedLedger().getConfig().getLedgerOffloader();
+                            if (!(offloader instanceof NullLedgerOffloader)) {
+                                try {
+                                    List<Metrics> metrics =
+                                            new LedgerOffloaderMetrics(pulsar, true, namespace, topicName).generate();
```

Comment:
It's better to pass the LedgerOffloader to the LedgerOffloaderMetrics.
Then we can delete LedgerOffloaderMetrics#aggregate 96~103

Reply:
yep, I agree with you

Reply:
Negative, we cannot pass `LedgerOfflaoder` to `LedgerOffloaderMetrics` here, because in some situation, we need to aggregate metrics in `namespace` level.